dhis-2/dhis-api/src/main/java/org/hisp/dhis/user/UserService.java

```diff
@@ -891,4 +893,15 @@ void validateTwoFactorUpdate(boolean before, boolean after, User userToModify)
   boolean isEmailVerified(User currentUser);
 
   User getUserByVerificationToken(String token);
+
+  /**
+   * Method that retrieves all {@link User}s that have an entry for the {@link OrganisationUnit} in
+   * the given table
+   *
+   * @param orgUnitTable table to search (one of 'organisationUnits', 'dataViewOrganisationUnits',
+   *     'teiSearchOrganisationUnits')
+   * @param uid {@link OrganisationUnit} {@link UID} to match on
+   * @return matching {@link User}s
+   */
+  List<User> getUsersWithOrgUnit(@Nonnull String orgUnitTable, @Nonnull UID uid);
```

Comment:
I don't think we should have names of tables escape the store. That is one of the main points to have a store abstraction. Maybe use an enum? 
While these 3 properties are backed by mapping tables in the DB in the object model these are properties so the wording should be accordingly.

Reply:
enum added